addons/point_of_sale/models/pos_session.py

```diff
@@ -206,6 +206,8 @@ def create(self, values):
                 'user_id': self.env.user.id,
                 'name': pos_name,
             }
+            last_session = self.env['pos.session'].search([('config_id', '=', config_id)], limit=1)
```

Comment:
maybe we could limit this when we actually have `pos_config.cash_control` active

Reply:
Yes. It is done.

Reply:
I was thinking something more like 
```
if pos_config.cash_control:
    last_session = self.env['pos.session'].search([('config_id', '=', config_id)], limit=1)
    ctx['previous_pos_session_id'] = last_session.id if last_session else False
```
so we fallback to the old compute method and the previous_statement_id is untouched

Reply:
If we fallback to the old compute method, we are exposed to the original issue of the incorrect `previous_statement_id`.
Even if we will not see the issue as we don't have cash control, the field will be wrongly set in db.
If a statement is created for POS session, it should use the override compute method.

Reply:
Sounds like a good point, but I'm not an expert of the bank statement functionality so I'll pass the last word to @nim-odoo